Frameworks/CoreFoundation/Preferences.subproj/CFPreferences.c

```diff
@@ -182,19 +182,24 @@ static CFURLRef _preferencesDirectoryForUserHostSafetyLevel(CFStringRef userName
 #if DEPLOYMENT_TARGET_WINDOWS
 
 CFURLRef url = NULL;
+Boolean success = false;
 
 CFMutableStringRef completePath = _CFCreateApplicationRepositoryPath(alloc, CSIDL_APPDATA);
 if (completePath) {
     // append "Preferences\" and make the CFURL
     CFStringAppend(completePath, CFSTR("Preferences\\"));
     url = CFURLCreateWithFileSystemPath(alloc, completePath, kCFURLWindowsPathStyle, true);
```

Comment:
Hi, @bbowman .

> I also don't quite understand the issue and what the fix is here. Can you elaborate?

The changes I made are fix for the issue [#1613 ](https://github.com/Microsoft/WinObjC/issues/1613). In my last comment to the issue 1613 I mentioned that I found the [solution](https://jongampark.wordpress.com/2009/06/24/be-careful-with-cfstringgetcstringptr-and-its-companion-calls/), which I applied in "_preferencesDirectoryForUserHostSafetyLevel" in CFPreferences.c  . But then I changed it to the implementation, which I submitted as PR, because I thought it was more relevant.